PublicMapping/districtbuilder
# Change topojson serialization format

## Overview

Upgrading to Node.js 16 in #602 caused a regression in data loading - the maximum size a of string was was reduced from ~1Gb to ~512Mb, which meant we could no longer load our two largest regions by using `JSON.parse(buffer.toString())`

This PR takes a multi-step approach to tackling this problem:
 - I've switched to serialzing the TopoJSON using the v8 binary serialization format: https://github.com/PublicMapping/districtbuilder/commit/c2a0dcb7ff876364e90f534b087b5c91ea9757bc, https://github.com/PublicMapping/districtbuilder/commit/d17738ea29fee0f69c1a90df1e79efec4669dc3e & https://github.com/PublicMapping/districtbuilder/commit/dc52da87c585429013c92c9bd4b7833d107c3f2a
 - Added a command to download the existing TopoJSON files (I used streaming JSON so we can load the now too large files w/o needing to downgrade Node temporarily) https://github.com/PublicMapping/districtbuilder/commit/f8262c278b49a46eb3c1e7e943207ac546b580ec
    - I've already applied this command on all the files used on staging as well as those referenced by `load-dev-data`, but I still need to update production, and any one-off regions will no longer work unless updated
   
After testing this branch on staging, I encountered a lot of resource contention and a large number of layers failed to load the first time, so I added https://github.com/PublicMapping/districtbuilder/commit/807dc9d57910850ca2acb3dd6ba5500b7999d419 & https://github.com/PublicMapping/districtbuilder/commit/e8ea4f103eb3120ea85728ab92e8dbc0e4628c29 to make the `TopologyService` more robust.

### Checklist


### Demo

Updated healthcheck, showing which layers are loading / pending.
![image](https://user-images.githubusercontent.com/4432106/148586493-8a4f323c-7769-4558-8f5f-a0d5b9bdd1ad.png)

### Notes

I was hopeful this would decrease our load times, but it hasn't done so to any meaningful degree. We did buy a lot of headroom in terms of TopoJSON file size - the max file size for [`v8.deserialize` is ~4GB](https://nodejs.org/api/v8.html#v8serializevalue).

## Testing Instructions

Try running `./scripts/manage serialize-topojson ...` locally, and make sure you can run the server

I tested this against staging, and was able to get staging to load successfully.

Closes #1095
